### PR TITLE
Added phone number length validation on client

### LINKED ISSUE #4433: Generic error message when checkout fails due to an overlength phone number
### Describe the bug

Checkout fails with 1 of 2 error messages when attempting to use a phone number longer than 20 characters, e.g. `123456789012345678901` (21 characters).

The error message is either:

>We're not able to process this request. Please refresh the page and try again.

Screen Shot 2022-07-13 at 12 53 15 pm

or

>Error: Invalid string: 1234567890...2345678901; must be at most 20 characters

Screen Shot 2022-07-13 at 12 53 04 pm

The first error message is problematic as it doesn't provide enough information to the customer about what the error is and how to rectify it.

The second error message is better because it references the invalid phone number in the error message. As such, the customer will likely be able to identify the problem and rectify it. It could be further improved by changing the wording `Error: Invalid string` to something like `Error: Invalid phone number`.

### To Reproduce

There are a few ways to reproduce this, see below:

**UPE enabled, new payment method**

1. Add a product to your cart and navigate to checkout.
2. Fill in valid billing details.
3. Use `123456789012345678901` for the billing phone number.
4. If prompted, click the "Use a new payment method" radio button and enter valid payment details.
5. Click the "Place order" button.
6. Observe that the transaction fails and you are presented with the following error message:
>Error: Invalid string: 1234567890...2345678901; must be at most 20 characters

**UPE enabled, saved payment method**

You must have an existing saved payment method to follow these steps.

1. Add a product to your cart and navigate to checkout.
2. Fill in valid billing details.
3. Use `123456789012345678901` for the billing phone number.
4. When prompted, select an existing payment method, e.g. `Visa ending in 4242`.
5. Click the "Place order" button.
6. Observe that the transaction fails and you are presented with the following error message:
>We're not able to process this request. Please refresh the page and try again.

### Actual behavior
Checkout fails with 1 of 2 error messages when attempting to use a phone number longer than 20 characters.

### Expected behavior
Clear and actionable error messages when checkout fails when attempting to use a phone number longer than 20 characters.

### Additional context

Issue observed in - 5336099-zd-woothemes

### PR DESCRIPTION
**Pull request by zmaglica**
Fixes #4433 

#### Changes proposed in this Pull Request

When a customer attempts to use a phone number longer than 20 characters, the error message was generic, and it wasn't clear to the customer what the error was and how to rectify it. This PR addresses this problem by validating phone number length on the client and if the phone number is over 20 characters, the error with message will be shown with detailed explanation.

#### Testing instructions

There are a few ways to test this, see below:

**UPE enabled, new payment method**

1. Add a product to your cart and navigate to checkout.
2. Fill in valid billing details.
3. Use `123456789012345678901` for the billing phone number.
4. If prompted, click the "Use a new payment method" radio button and enter valid payment details.
5. Click the "Place order" button.
6. Observe that the transaction fails and you are presented with the following error message:
>Error:  Invalid phone number

**UPE enabled, saved payment method**

You must have an existing saved payment method to follow these steps.

1. Add a product to your cart and navigate to checkout.
2. Fill in valid billing details.
3. Use `123456789012345678901` for the billing phone number.
4. When prompted, select an existing payment method, e.g. `Visa ending in 4242`.
5. Click the "Place order" button.
6. Observe that the transaction fails and you are presented with the following error message:
>Error:  Invalid phone number

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Review by jessy-p — COMMENTED** [review]
Can you please check for non-UPE checkout, it is not working.

File: client/checkout/classic/index.js [hunk 1]
```diff
@@ -303,6 +308,14 @@ jQuery( function ( $ ) {
 	 * @param {Object} paymentMethod Payment method object.
 	 */
 	const handleOrderPayment = ( $form, { id } ) => {
+		if (
```
Comment (naman03malhotra) [inline client/checkout/classic/index.js]: Looks like a repeated piece of code to me, can this function be separated in a `ùtil.js` file and used across? That way, we can isolate the function's behavior and write the test only once?

File: client/checkout/classic/upe.js [hunk 2]
```diff
@@ -408,6 +408,16 @@ jQuery( function ( $ ) {
 			);
 			return false;
 		}
+		if (
```
Comment (naman03malhotra) [inline client/checkout/classic/upe.js]: Can we write unit tests for these changes?

**Review by naman03malhotra — DISMISSED** [review]
I've left some comments related to code quality refactoring and tests. I am yet to test the PR.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5762` or branch name `fix/4433-improve-error-message-on-invalid-phone` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/4433-improve-error-message-on-invalid-phone) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: c07c88d67bc7b1175803ee1ddd4d07b067cbd244
- Build time: 2023-03-30 12:12:48 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by jessy-p — COMMENTED** [review]
Appear to be no changes in `upe.js` or `upe-split.js`, you can checkout the files from trunk to make your PR cleaner.

**Review by jessy-p — COMMENTED** [review]
With Split UPE and using a new card, I get a different error than the one in this PR 
image
This is the error on attempting to check out with Giropay as well.

**Comment by jessy-p** [comment]
Tested different checkout flows and working as expected.

**Review by jessy-p — APPROVED** [review]

**Review by jessy-p — APPROVED** [review]

File: client/checkout/classic/upe.js (hunk 2 above)
Comment (zmaglica) [reply client/checkout/classic/upe.js]: Unit tests have been added.

**Pull request merged**